src/alchemlyb/parsing/amber.py

```diff
@@ -208,6 +210,11 @@ def file_validation(outfile):
                 have_mbar = False
             else:
                 mbar_nlambda = len(mbar_lambdas)
+                if mbar_nlambda != mbar_states:
+                    logger.warning(
+                        "The number of lambda windows read (%s)"
+                        " is differemt from what expected (%d)",
+                        ','.join(mbar_lambdas), mbar_states)
```

Comment:
Shouldn't this be an exception?

Reply:
I think so, I just wrote a warning because with the previous version we were not rising exceptions when this section wasn't parsed correctly, but I agree we should.

Reply:
Let's raise an exception here. Use `ValueError` whenever something is wrong in the input file because we use `ValueError` in the NAMD parser in the same context.

Please add an appropriate test (copy & modify the test where you check for the logger messages — btw, nicely written tests! — and check `with pytest.raises(ValueError, match="The number of lambda windows read")`.

Reply:
Done and test added (thanks for the comment :-) )!